editor/src/components/canvas/dom-walker.ts

```diff
@@ -408,9 +413,17 @@ function collectMetadata(
   )
 
   return pathsForElement.map((path) => {
-    const filteredChildPaths = unfilteredChildrenPaths.filter((childPath) =>
-      TP.pathsEqual(path, TP.parentPath(childPath)),
-    )
+    let filteredChildPaths: InstancePath[] = []
+    let filteredRootElements: InstancePath[] = []
```

Comment:
hmmmmmm, hmmmmmmmmmmmmm, `MetadataUtils.getAllChildrenIncludingUnfurledFocusedComponents` now filters through allPaths to find `TP.scenePathsEqual(TP.scenePathPartOfTemplatePath(p), matchingFocusPath),` 

https://github.com/concrete-utopia/utopia/blob/e2e11c462a8ce61508a4b84d2e417db9008a8619/editor/src/core/model/element-metadata-utils.ts#L805

If I understand this correctly, instead of the (slightly expensive) array creationg and array filtering, it could use the rootElements property

Reply:
:thinking: that _sounds_ correct

Reply:
interesting! this is actually more useful than I thought – however, I realized just now that you are kinda cheating with Scenes, the root element of a scene is a... div? and not really the "target component" of the scene – this is not a problem as next week we are going to change how a Scene works, but sneaky sneaky